opengrok-web/src/main/java/org/opengrok/web/util/FileUtil.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ */
+
+package org.opengrok.web.util;
+
+import org.opengrok.indexer.configuration.RuntimeEnvironment;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+
+public class FileUtil {
+
+    private static final RuntimeEnvironment env = RuntimeEnvironment.getInstance();
+
+    // private to enforce static
+    private FileUtil() {
+    }
+
+    public static class NoFileException extends Exception {
+        private static final long serialVersionUID = 1L;
+
+        public NoFileException(String message) {
+            super(message);
+        }
+    }
+
+    public static class NoPathException extends Exception {
+        private static final long serialVersionUID = 1L;
+
+        public NoPathException(String message) {
+            super(message);
+        }
+    }
+
+    /**
+     * @param path path relative to source root
+     * @return file object corresponding to the file under source root
+     * @throws NoFileException
+     * @throws NoPathException
+     */
+    public static File getFile(String path, HttpServletResponse response) throws NoPathException, NoFileException {
+        if (path == null) {
+            throw new NoPathException("Missing path parameter");
+        }
+
+        File file = new File(env.getSourceRootFile(), path);
+        if (!file.isFile()) {
+            throw new NoFileException("File not found");
```

Comment:
There is already FileNotFound (or similar) in java core

Reply:
aha ! thanks.